main.go

```diff
@@ -10,21 +10,39 @@ import (
 )
 
 type ProxyConfig struct {
+	// Port on whch the proxy is listening on
 	Port int `env:"PORT" envDefault:"3000"`
 
 	// Proxy target
 	RemoteAddress string `env:"REMOTE_ADDRESS"`
 
-	// InputValidation enables the proxy to validate the incoming proxy
-	// requests using
+	// InputValidation enables the proxy to validate / authenticate the
+	// incoming proxy requests using
 	InputValidation bool `env:"INPUT_VALIDATION"`
-	// Allowed Public identities passed in as comma separated b64 encoded
-	// pubkeys
+
+	// one of: [ "ecdsasignatures", "basicauth", "none" ]
+	AuthenticationScheme string `env:"AUTHENTICATION_SCHEME" envDefault:"basicauth"`
+
+	// if AuthenticationScheme is basicauth then this string is used for
+	// determinig allowed user / password pairs.
+	// NOTE: the string format is: user1:password1,user2:password2
+	AllowedBasicAuthUserString string `env:"ALLOWED_USERS_STRING" envDefault:""`
+
+	// If AuthenticationScheme is ecdsasigning then this string is used for
+	// determining allowed public identities
+	// NOTE: the string format is a comma separated b64 encoded pubkeys
 	AllowedIDs string `env:"ALLOWED_IDS"`
```

Comment:
Since we have several conditionally used envs, can we throw if one is supplied but not used? E.g., using `basicauth` and supplying `ALLOWED_IDS`?

Reply:
We could, however, the only things I see against is:
- If an operator will transition from one to the other, he'd need to remember to remove the env (from configs) to do that, otherwise the container will just throw and stop. Also switching might be more involved?
- We could potentially offer multiple authentication methods in the future (for example, signed requests for ingesting logs, yet people who want to use kibana can use HTTP Basic Auth)

Reply:
I see that first point as a feature not a bug. They _should_ be removed, and the new ones added. Otherwise it'll run but it'll probably not do what they think it's doing. As for point 2, that would require a wholesale refactor of these flags and we can deal with that then.